### DISCUSSION TITLE
想请教下，为何渲染部分的代码看起来大量的依赖了Vulkan的数据类型？

### DISCUSSION THREAD

**Discussion by at2021** [BoomingTech/Piccolo#299] [Q&A] [answered]
按理说不是应该把图形相关的API都封装成RHI的接口，上层的渲染代码不需要知道当前用的API吗？看现在的代码，如果想要切换到其他图像API的话岂不是有非常大量的工作需要做？

**Comment by Ol6rin** [comment, marked as answer]
确实存在这个问题。Piccolo在设计之初时为了降低理解难度做了简化，使用了Vulkan一种API达到最大的跨平台兼容性。
我们内部也在整理RHI抽象层，也在讨论后面的架构调整。
如果感兴趣参与后续RHI的开发，请关注Issue #333
